plugin/ball_plugin.cpp

```diff
@@ -0,0 +1,95 @@
+#include <chrono>
+#include <gazebo/common/common.hh>
+#include <gazebo/gazebo.hh>
+#include <gazebo/physics/physics.hh>
+#include <gazebo/sensors/sensors.hh>
+
+#include "clock_utils.h"
+#include "message/platform/gazebo/Ball.pb.h"
+#include "nuclear_network.h"
+
+namespace gazebo {
+class NUbotsBallPlugin : public ModelPlugin {
+
+public:
+    // Inherit constructors
+    using ModelPlugin::ModelPlugin;
+
+    // Make sure to unbind the reaction handle on destruction
+    virtual ~NUbotsBallPlugin() {
+        // Unbind handles on destruction
+        for (auto& h : handles) {
+            h.unbind();
+        }
+    }
+
+    /**
+     * The load function is called by Gazebo when the plugin is inserted into simulation
+     * @param _model A pointer to the model that this plugin is attached to
+     * @param _sdf   A pointer to the plugin's SDF element
+     */
+    void Load(physics::ModelPtr _model, sdf::ElementPtr _sdf) {
+
+        // Output a debug line
+        gzdbg << "Attaching a ball plugin to model [" << _model->GetName() << "]" << std::endl;
+
+        // Store the model pointer for convenience
+        model = _model;
```

Comment:
In some of the other code before when you replaced `_model` with `model` you forgot to update this, so if you do remember to change this too